File: create_guide.js
const { Document, Packer, Paragraph, TextRun, PageBreak, HeadingLevel, AlignmentType, BorderStyle, WidthType } = require('docx');
const fs = require('fs');

const doc = new Document({
  styles: {
    default: {
      document: {
        run: { font: "Arial", size: 22 }
      }
    },
    paragraphStyles: [
      {
        id: "Heading1",
        name: "Heading 1",
        basedOn: "Normal",
        next: "Normal",
        quickFormat: true,
        run: { size: 32, bold: true, font: "Arial", color: "1a3a52" },
        paragraph: { spacing: { before: 240, after: 120 }, outlineLevel: 0 }
      },
      {
        id: "Heading2",
        name: "Heading 2",
        basedOn: "Normal",
        next: "Normal",
        quickFormat: true,
        run: { size: 28, bold: true, font: "Arial", color: "2c5aa0" },
        paragraph: { spacing: { before: 200, after: 100 }, outlineLevel: 1 }
      }
    ]
  },
  sections: [{
    properties: {
      page: {
        size: { width: 12240, height: 15840 },
        margin: { top: 1440, right: 1440, bottom: 1440, left: 1440 }
      }
    },
    children: [
      // PORTADA
      new Paragraph({ text: "", spacing: { line: 800 } }),
      new Paragraph({ text: "", spacing: { line: 800 } }),
      new Paragraph({ text: "", spacing: { line: 800 } }),

      new Paragraph({
        text: "GUÍA DE INSTALACIÓN",
        alignment: AlignmentType.CENTER,
        spacing: { after: 200 },
        style: "Heading1"
      }),

      new Paragraph({
        text: "Sistema Jurídico v1.0.0",
        alignment: AlignmentType.CENTER,
        spacing: { after: 400 },
        style: "Heading2"
      }),

      new Paragraph({
        text: "Windows y macOS",
        alignment: AlignmentType.CENTER,
        spacing: { after: 800 },
        run: { size: 24 }
      }),

      new Paragraph({ text: "", spacing: { line: 800 } }),
      new Paragraph({ text: "", spacing: { line: 800 } }),

      new Paragraph({
        text: "Marzo 2026",
        alignment: AlignmentType.CENTER,
        run: { size: 24 }
      }),

      new Paragraph({ children: [new PageBreak()] }),

      // TABLA DE CONTENIDOS
      new Paragraph({
        text: "TABLA DE CONTENIDOS",
        style: "Heading2"
      }),

      new Paragraph({ text: "1. Requisitos Previos", spacing: { after: 100 } }),
      new Paragraph({ text: "2. Instalación en Windows", spacing: { after: 100 } }),
      new Paragraph({ text: "3. Instalación en macOS", spacing: { after: 100 } }),
      new Paragraph({ text: "4. Errores Comunes y Soluciones", spacing: { after: 100 } }),
      new Paragraph({ text: "5. Preguntas Frecuentes", spacing: { after: 100 } }),
      new Paragraph({ text: "6. Soporte Técnico", spacing: { after: 400 } }),

      new Paragraph({ children: [new PageBreak()] }),

      // SECCIÓN 1
      new Paragraph({
        text: "1. REQUISITOS PREVIOS",
        style: "Heading2"
      }),

      new Paragraph({
        text: "Para ambos sistemas:",
        spacing: { after: 100 },
        run: { bold: true }
      }),

      new Paragraph({ text: "Mínimo 500 MB de espacio libre en disco", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Conexión a Internet (solo para descarga)", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Permisos de administrador en tu equipo", spacing: { after: 200 }, indent: { left: 360 } }),

      new Paragraph({
        text: "Windows:",
        spacing: { after: 100 },
        run: { bold: true }
      }),

      new Paragraph({ text: "Windows 10 o superior (64 bits)", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "No requiere .NET Framework adicional", spacing: { after: 200 }, indent: { left: 360 } }),

      new Paragraph({
        text: "macOS:",
        spacing: { after: 100 },
        run: { bold: true }
      }),

      new Paragraph({ text: "macOS 10.15 (Catalina) o superior", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Procesador Intel o Apple Silicon (M1/M2/M3)", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Autorización para ejecutar apps de desarrolladores no identificados", spacing: { after: 400 }, indent: { left: 360 } }),

      new Paragraph({ children: [new PageBreak()] }),

      // SECCIÓN 2
      new Paragraph({
        text: "2. INSTALACIÓN EN WINDOWS",
        style: "Heading2"
      }),

      new Paragraph({
        text: "Paso 1: Descargar el Instalador",
        spacing: { after: 100 },
        run: { bold: true }
      }),

      new Paragraph({ text: 'Descarga el archivo: "Sistema Jurídico Setup 1.0.0.exe" desde GitHub Actions', spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Guarda el archivo en tu escritorio o carpeta de descargas", spacing: { after: 200 }, indent: { left: 360 } }),

      new Paragraph({
        text: "Paso 2: Ejecutar el Instalador",
        spacing: { after: 100 },
        run: { bold: true }
      }),

      new Paragraph({ text: "Haz doble clic sobre el archivo .exe", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Si aparece Control de Cuentas de Usuario (UAC): Haz clic en SÍ", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Sigue el asistente: selecciona carpeta (C:\\Archivos de programa\\), marca opciones deseadas", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Haz clic en INSTALAR", spacing: { after: 200 }, indent: { left: 360 } }),

      new Paragraph({
        text: "Paso 3: Completar la Instalación",
        spacing: { after: 100 },
        run: { bold: true }
      }),

      new Paragraph({ text: "Espera a que finalice", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Haz clic en FINALIZAR", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "La aplicación se lanzará automáticamente", spacing: { after: 200 }, indent: { left: 360 } }),

      new Paragraph({
        text: "Opción: Usar la Versión Portable",
        spacing: { after: 100 },
        run: { bold: true }
      }),

      new Paragraph({ text: 'Descarga: "SistemaJuridico-portable-1.0.0.exe"', spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Haz doble clic para ejecutar directamente (sin instalación)", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "No requiere permisos de administrador", spacing: { after: 400 }, indent: { left: 360 } }),

      new Paragraph({ children: [new PageBreak()] }),

      // SECCIÓN 3
      new Paragraph({
        text: "3. INSTALACIÓN EN macOS",
        style: "Heading2"
      }),

      new Paragraph({
        text: "Paso 1: Descargar",
        spacing: { after: 100 },
        run: { bold: true }
      }),

      new Paragraph({ text: 'Descarga el archivo: "Sistema Jurídico-1.0.0.dmg" desde GitHub Actions', spacing: { after: 200 }, indent: { left: 360 } }),

      new Paragraph({
        text: "Paso 2: Montar la Imagen",
        spacing: { after: 100 },
        run: { bold: true }
      }),

      new Paragraph({ text: "Haz doble clic sobre el archivo .dmg", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Se abrirá una ventana con la aplicación", spacing: { after: 200 }, indent: { left: 360 } }),

      new Paragraph({
        text: "Paso 3: Instalar",
        spacing: { after: 100 },
        run: { bold: true }
      }),

      new Paragraph({ text: 'Arrastra el icono "Sistema Jurídico" a la carpeta Applications', spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Espera 1-2 minutos", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Cierra la ventana", spacing: { after: 200 }, indent: { left: 360 } }),

      new Paragraph({
        text: "Paso 4: Ejecución Inicial (IMPORTANTE)",
        spacing: { after: 100 },
        run: { bold: true }
      }),

      new Paragraph({ text: "Primera vez solamente: macOS bloqueará la ejecución por motivos de seguridad.", spacing: { after: 100 }, run: { italics: true } }),

      new Paragraph({ text: "Abre Finder → Aplicaciones", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Haz CLIC DERECHO sobre \'Sistema Jurídico\'", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Selecciona ABRIR", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Haz clic en ABRIR en la confirmación", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Próximas veces: simplemente haz doble clic normalmente", spacing: { after: 200 }, indent: { left: 360 } }),

      new Paragraph({
        text: "Paso 5: Desmontar",
        spacing: { after: 100 },
        run: { bold: true }
      }),

      new Paragraph({ text: "En Finder, haz clic en el icono expulsar junto a la imagen", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Opcionalmente, elimina el archivo .dmg de Descargas", spacing: { after: 400 }, indent: { left: 360 } }),

      new Paragraph({ children: [new PageBreak()] }),

      // SECCIÓN 4
      new Paragraph({
        text: "4. ERRORES COMUNES Y SOLUCIONES",
        style: "Heading2"
      }),

      new Paragraph({
        text: 'ERROR: "No se puede verificar el desarrollador" (macOS)',
        spacing: { after: 100 },
        run: { bold: true }
      }),

      new Paragraph({ text: "Cuando aparece: Primera ejecución", spacing: { after: 100 }, run: { bold: true } }),
      new Paragraph({ text: "Solución: Clic DERECHO en la app → Abrir → Abrir", spacing: { after: 200 }, indent: { left: 360 } }),

      new Paragraph({
        text: 'ERROR: "No tiene permiso para acceder" (Windows)',
        spacing: { after: 100 },
        run: { bold: true }
      }),

      new Paragraph({ text: "Cuando aparece: Instalación en carpeta protegida", spacing: { after: 100 }, run: { bold: true } }),
      new Paragraph({ text: "Desinstala completamente", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Clic DERECHO en .exe → Ejecutar como administrador", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Instala en C:\\Archivos de programa\\", spacing: { after: 200 }, indent: { left: 360 } }),

      new Paragraph({
        text: 'ERROR: "La aplicación no responde" (ambos sistemas)',
        spacing: { after: 100 },
        run: { bold: true }
      }),

      new Paragraph({ text: "Cierra la aplicación", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Elimina carpeta de datos: Windows: C:\\Users\\[Usuario]\\AppData\\Local\\Sistema Jurídico\\", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "macOS: ~/Library/Application Support/Sistema Jurídico/", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Reinicia", spacing: { after: 200 }, indent: { left: 360 } }),

      new Paragraph({
        text: 'ERROR: "No se puede conectar a la base de datos" (ambos)',
        spacing: { after: 100 },
        run: { bold: true }
      }),

      new Paragraph({ text: "Causa: Base de datos corrupta", spacing: { after: 100 }, run: { bold: true } }),
      new Paragraph({ text: "Cierra la aplicación", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Elimina: biblioteca.db", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "  Windows: %APPDATA%\\Sistema Jurídico\\prisma\\biblioteca.db", spacing: { after: 80 }, indent: { left: 720 } }),
      new Paragraph({ text: "  macOS: ~/Library/Application Support/Sistema Jurídico/prisma/biblioteca.db", spacing: { after: 80 }, indent: { left: 720 } }),
      new Paragraph({ text: "Reinicia (se recreará automáticamente)", spacing: { after: 400 }, indent: { left: 360 } }),

      new Paragraph({ children: [new PageBreak()] }),

      // SECCIÓN 5
      new Paragraph({
        text: "5. PREGUNTAS FRECUENTES",
        style: "Heading2"
      }),

      new Paragraph({
        text: "¿Necesito conexión a Internet?",
        spacing: { after: 80 },
        run: { bold: true }
      }),
      new Paragraph({ text: "Solo para descarga inicial. Funciona sin conexión después.", spacing: { after: 200 } }),

      new Paragraph({
        text: "¿Cómo desinstalo?",
        spacing: { after: 80 },
        run: { bold: true }
      }),
      new Paragraph({ text: "Windows - Panel de Control → Desinstalar un programa. macOS - Arrastra a Papelera", spacing: { after: 200 } }),

      new Paragraph({
        text: "¿Dónde se guardan los datos?",
        spacing: { after: 80 },
        run: { bold: true }
      }),
      new Paragraph({ text: "Localmente en tu equipo (Windows: %APPDATA%, macOS: ~/Library)", spacing: { after: 200 } }),

      new Paragraph({
        text: "¿Puedo ejecutar dos instancias?",
        spacing: { after: 80 },
        run: { bold: true }
      }),
      new Paragraph({ text: "No, solo una por usuario", spacing: { after: 200 } }),

      new Paragraph({
        text: "¿Qué hago si olvido mis credenciales?",
        spacing: { after: 80 },
        run: { bold: true }
      }),
      new Paragraph({ text: "Contacta al administrador del sistema", spacing: { after: 200 } }),

      new Paragraph({
        text: "¿Es seguro?",
        spacing: { after: 80 },
        run: { bold: true }
      }),
      new Paragraph({ text: "Sí, todos los datos se almacenan localmente sin envío a servidores externos", spacing: { after: 400 } }),

      new Paragraph({ children: [new PageBreak()] }),

      // SECCIÓN 6
      new Paragraph({
        text: "6. SOPORTE TÉCNICO",
        style: "Heading2"
      }),

      new Paragraph({
        text: "Si experimentas problemas:",
        spacing: { after: 100 },
        run: { bold: true }
      }),

      new Paragraph({ text: "Verifica tu conexión de Internet", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Reinicia tu equipo", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "Desinstala y reinstala la aplicación", spacing: { after: 200 }, indent: { left: 360 } }),

      new Paragraph({
        text: "Para soporte adicional:",
        spacing: { after: 100 },
        run: { bold: true }
      }),

      new Paragraph({ text: "Email: angelixvrobles1234@outlook.com", spacing: { after: 80 }, indent: { left: 360 } }),
      new Paragraph({ text: "GitHub: https://github.com/AngelixVrobles/sistema-juridico", spacing: { after: 300 }, indent: { left: 360 } }),

      new Paragraph({
        text: "Información del Documento",
        spacing: { after: 100 },
        run: { bold: true }
      }),

      new Paragraph({ text: "Última actualización: Marzo 2026", spacing: { after: 80 } }),
      new Paragraph({ text: "Aplicación: Sistema Jurídico v1.0.0", spacing: { after: 80 } }),
      new Paragraph({ text: "Plataformas soportadas: Windows 10+ (64 bits), macOS 10.15+", spacing: { after: 300 } }),

      new Paragraph({
        text: "¡Gracias por usar Sistema Jurídico!",
        alignment: AlignmentType.CENTER,
        spacing: { after: 100 },
        style: "Heading2"
      })
    ]
  }]
});

Packer.toBuffer(doc).then(buffer => {
  fs.writeFileSync("Guia_Instalacion_SistemaJuridico.docx", buffer);
  console.log("✅ Documento DOCX creado exitosamente: Guia_Instalacion_SistemaJuridico.docx");
});

File: create_guide.py
from reportlab.lib.pagesizes import letter
from reportlab.lib.styles import getSampleStyleSheet, ParagraphStyle
from reportlab.lib.units import inch
from reportlab.platypus import SimpleDocTemplate, Paragraph, Spacer, PageBreak
from reportlab.lib import colors
from reportlab.lib.enums import TA_CENTER, TA_JUSTIFY

# Crear el documento PDF
pdf_path = "Guia_Instalacion_SistemaJuridico.pdf"
doc = SimpleDocTemplate(pdf_path, pagesize=letter, topMargin=0.5*inch, bottomMargin=0.5*inch)

# Estilos
styles = getSampleStyleSheet()
title_style = ParagraphStyle(
    'CustomTitle',
    parent=styles['Heading1'],
    fontSize=24,
    textColor=colors.HexColor('#1a3a52'),
    spaceAfter=6,
    alignment=TA_CENTER,
    fontName='Helvetica-Bold'
)

heading2_style = ParagraphStyle(
    'CustomHeading2',
    parent=styles['Heading2'],
    fontSize=14,
    textColor=colors.HexColor('#2c5aa0'),
    spaceAfter=12,
    spaceBefore=12,
    fontName='Helvetica-Bold'
)

heading3_style = ParagraphStyle(
    'CustomHeading3',
    parent=styles['Heading3'],
    fontSize=12,
    textColor=colors.HexColor('#34568B'),
    spaceAfter=6,
    fontName='Helvetica-Bold'
)

body_style = ParagraphStyle(
    'CustomBody',
    parent=styles['Normal'],
    fontSize=10,
    alignment=TA_JUSTIFY,
    spaceAfter=8,
    leading=14
)

# Contenido del documento
story = []

# Portada
story.append(Spacer(1, 1.5*inch))
story.append(Paragraph("GUÍA DE INSTALACIÓN", title_style))
story.append(Paragraph("Sistema Jurídico v1.0.0", styles['Heading2']))
story.append(Spacer(1, 0.5*inch))
story.append(Paragraph("Windows y macOS", styles['Normal']))
story.append(Spacer(1, 2*inch))
story.append(Paragraph("Marzo 2026", styles['Normal']))
story.append(PageBreak())

# Tabla de contenidos
story.append(Paragraph("TABLA DE CONTENIDOS", heading2_style))
toc_items = [
    "1. Requisitos Previos",
    "2. Instalación en Windows",
    "3. Instalación en macOS",
    "4. Errores Comunes y Soluciones",
    "5. Preguntas Frecuentes",
    "6. Soporte Técnico"
]
for item in toc_items:
    story.append(Paragraph(item, body_style))
story.append(PageBreak())

# 1. REQUISITOS PREVIOS
story.append(Paragraph("1. REQUISITOS PREVIOS", heading2_style))

story.append(Paragraph("<b>Para ambos sistemas:</b>", heading3_style))
requisitos_ambos = [
    "Mínimo 500 MB de espacio libre en disco",
    "Conexión a Internet (solo para descarga)",
    "Permisos de administrador en tu equipo"
]
for req in requisitos_ambos:
    story.append(Paragraph("• " + req, body_style))

story.append(Paragraph("<b>Windows:</b>", heading3_style))
story.append(Paragraph("• Windows 10 o superior (64 bits)", body_style))
story.append(Paragraph("• No requiere .NET Framework adicional", body_style))

story.append(Paragraph("<b>macOS:</b>", heading3_style))
story.append(Paragraph("• macOS 10.15 (Catalina) o superior", body_style))
story.append(Paragraph("• Procesador Intel o Apple Silicon (M1/M2/M3)", body_style))
story.append(Paragraph("• Autorización para ejecutar apps de desarrolladores no identificados", body_style))
story.append(Spacer(1, 0.2*inch))
story.append(PageBreak())

# 2. INSTALACIÓN EN WINDOWS
story.append(Paragraph("2. INSTALACIÓN EN WINDOWS", heading2_style))

story.append(Paragraph("<b>Paso 1: Descargar el Instalador</b>", heading3_style))
story.append(Paragraph("1. Descarga el archivo: <b>Sistema Jurídico Setup 1.0.0.exe</b> desde GitHub Actions", body_style))
story.append(Paragraph("2. Guarda el archivo en tu escritorio o carpeta de descargas", body_style))

story.append(Paragraph("<b>Paso 2: Ejecutar el Instalador</b>", heading3_style))
story.append(Paragraph("1. Haz doble clic sobre el archivo .exe", body_style))
story.append(Paragraph("2. Si aparece una ventana de Control de Cuentas de Usuario (UAC): Haz clic en SI", body_style))
story.append(Paragraph("3. Sigue el asistente de instalación:", body_style))
story.append(Paragraph("   - Selecciona la carpeta de instalación (recomendado: C:\\Archivos de programa\\)", body_style))
story.append(Paragraph("   - Marca las opciones deseadas (acceso directo en escritorio, menú de inicio)", body_style))
story.append(Paragraph("   - Haz clic en INSTALAR", body_style))

story.append(Paragraph("<b>Paso 3: Completar la Instalación</b>", heading3_style))
story.append(Paragraph("• Espera a que finalice el proceso", body_style))
story.append(Paragraph("• Haz clic en FINALIZAR", body_style))
story.append(Paragraph("• La aplicación se lanzará automáticamente", body_style))

story.append(Paragraph("<b>Opción: Usar la Versión Portable</b>", heading3_style))
story.append(Paragraph("1. Descarga: <b>SistemaJuridico-portable-1.0.0.exe</b>", body_style))
story.append(Paragraph("2. Haz doble clic para ejecutar directamente (sin instalación)", body_style))
story.append(Paragraph("3. No requiere permisos de administrador", body_style))
story.append(PageBreak())

# 3. INSTALACIÓN EN macOS
story.append(Paragraph("3. INSTALACIÓN EN macOS", heading2_style))

story.append(Paragraph("<b>Paso 1: Descargar la Imagen de Disco</b>", heading3_style))
story.append(Paragraph("1. Descarga el archivo: <b>Sistema Jurídico-1.0.0.dmg</b> desde GitHub Actions", body_style))
story.append(Paragraph("2. El archivo se descargará en tu carpeta de Descargas", body_style))

story.append(Paragraph("<b>Paso 2: Montar la Imagen</b>", heading3_style))
story.append(Paragraph("1. Haz doble clic sobre el archivo .dmg", body_style))
story.append(Paragraph("2. Se abrirá una ventana mostrando la aplicación Sistema Jurídico", body_style))
story.append(Paragraph("3. Verás una carpeta Applications (Aplicaciones)", body_style))

story.append(Paragraph("<b>Paso 3: Instalar la Aplicación</b>", heading3_style))
story.append(Paragraph("1. Arrastra el icono Sistema Jurídico a la carpeta Applications", body_style))
story.append(Paragraph("2. Espera a que se complete la copia (puede tomar 1-2 minutos)", body_style))
story.append(Paragraph("3. Cierra la ventana del instalador", body_style))

story.append(Paragraph("<b>Paso 4: Ejecución Inicial (IMPORTANTE)</b>", heading3_style))
story.append(Paragraph("Primera vez solamente: macOS bloqueará la ejecución por motivos de seguridad.", body_style))
story.append(Paragraph("1. Abre Finder y ve a Aplicaciones", body_style))
story.append(Paragraph("2. Localiza Sistema Jurídico", body_style))
story.append(Paragraph("3. Haz clic DERECHO sobre el icono", body_style))
story.append(Paragraph("4. Selecciona ABRIR en el menú", body_style))
story.append(Paragraph("5. Haz clic en ABRIR en la ventana de confirmación", body_style))
story.append(Paragraph("6. La aplicación se ejecutará", body_style))
story.append(Spacer(1, 0.1*inch))
story.append(Paragraph("Próximas veces: Simplemente haz doble clic normalmente", body_style))

story.append(Paragraph("<b>Paso 5: Desmontar la Imagen</b>", heading3_style))
story.append(Paragraph("1. En el Finder, busca Sistema Jurídico en la barra lateral", body_style))
story.append(Paragraph("2. Haz clic en el icono de expulsar junto a ella", body_style))
story.append(Paragraph("3. Opcionalmente, elimina el archivo .dmg de Descargas", body_style))
story.append(PageBreak())

# 4. ERRORES COMUNES
story.append(Paragraph("4. ERRORES COMUNES Y SOLUCIONES", heading2_style))

story.append(Paragraph("<b>ERROR: No se puede verificar el desarrollador (macOS)</b>", heading3_style))
story.append(Paragraph("<b>Cuando aparece:</b> Primera ejecución de la aplicación en macOS", body_style))
story.append(Paragraph("<b>Solucion rapida (recomendada):</b>", body_style))
story.append(Paragraph("Clic DERECHO en la aplicación → Abrir → Abrir", body_style))
story.append(Spacer(1, 0.15*inch))

story.append(Paragraph("<b>ERROR: No tiene permiso para acceder a este recurso (Windows)</b>", heading3_style))
story.append(Paragraph("<b>Cuando aparece:</b> Instalación en carpeta protegida", body_style))
story.append(Paragraph("1. Desinstala completamente la aplicación", body_style))
story.append(Paragraph("2. Haz clic DERECHO en el instalador .exe", body_style))
story.append(Paragraph("3. Selecciona EJECUTAR COMO ADMINISTRADOR", body_style))
story.append(Paragraph("4. Instala en: C:\\Archivos de programa\\", body_style))
story.append(Spacer(1, 0.15*inch))

story.append(Paragraph("<b>ERROR: La aplicación no responde (ambos sistemas)</b>", heading3_style))
story.append(Paragraph("1. Cierra la aplicación", body_style))
story.append(Paragraph("2. Elimina la carpeta de datos de usuario:", body_style))
story.append(Paragraph("   Windows: C:\\Users\\[TuUsuario]\\AppData\\Local\\Sistema Jurídico\\", body_style))
story.append(Paragraph("   macOS: ~/Library/Application Support/Sistema Jurídico/", body_style))
story.append(Paragraph("3. Reinicia la aplicación", body_style))
story.append(Spacer(1, 0.15*inch))

story.append(Paragraph("<b>ERROR: No se puede conectar a la base de datos (ambos sistemas)</b>", heading3_style))
story.append(Paragraph("<b>Causa:</b> Archivo de base de datos corrupto o permisos insuficientes", body_style))
story.append(Paragraph("1. Cierra la aplicación completamente", body_style))
story.append(Paragraph("2. Busca y elimina el archivo biblioteca.db:", body_style))
story.append(Paragraph("   Windows: %APPDATA%\\Sistema Jurídico\\prisma\\biblioteca.db", body_style))
story.append(Paragraph("   macOS: ~/Library/Application Support/Sistema Jurídico/prisma/biblioteca.db", body_style))
story.append(Paragraph("3. Reinicia la aplicación (se recreará automáticamente)", body_style))
story.append(PageBreak())

# 5. PREGUNTAS FRECUENTES
story.append(Paragraph("5. PREGUNTAS FRECUENTES", heading2_style))

faqs = [
    ("¿Necesito conexión a Internet?", "Solo para la descarga inicial. Una vez instalada, funciona sin conexión."),
    ("¿Puedo desinstalar la aplicación?", "Windows: Panel de Control → Desinstalar un programa → Sistema Jurídico. macOS: Arrastra la aplicación a la Papelera"),
    ("¿Donde se guardan mis datos?", "Windows: %APPDATA%\\Sistema Jurídico/. macOS: ~/Library/Application Support/Sistema Jurídico/"),
    ("¿Puedo ejecutar dos instancias?", "No. La aplicación solo se puede ejecutar una vez por usuario."),
    ("¿Qué hago si olvido mis credenciales?", "Contacta al administrador del sistema para reseteo de credenciales."),
    ("¿Es seguro usar esta aplicación?", "Si. Todos los datos se almacenan localmente sin envio a servidores externos.")
]

for pregunta, respuesta in faqs:
    story.append(Paragraph(f"<b>P: {pregunta}</b>", heading3_style))
    story.append(Paragraph(f"<b>R:</b> {respuesta}", body_style))
    story.append(Spacer(1, 0.1*inch))

story.append(PageBreak())

# 6. SOPORTE TÉCNICO
story.append(Paragraph("6. SOPORTE TECNICO", heading2_style))

story.append(Paragraph("Si experimentas problemas no listados aquí:", body_style))
story.append(Paragraph("1. Verifica tu conexión de Internet", body_style))
story.append(Paragraph("2. Reinicia tu equipo", body_style))
story.append(Paragraph("3. Desinstala y reinstala la aplicación", body_style))
story.append(Spacer(1, 0.2*inch))

story.append(Paragraph("<b>Para soporte adicional o reportar bugs:</b>", heading3_style))
story.append(Paragraph("Email: angelixvrobles1234@outlook.com", body_style))
story.append(Paragraph("GitHub: https://github.com/AngelixVrobles/sistema-juridico", body_style))
story.append(Spacer(1, 0.3*inch))

story.append(Paragraph("<b>Información del Documento</b>", heading3_style))
story.append(Paragraph("Última actualización: Marzo 2026", body_style))
story.append(Paragraph("Aplicación: Sistema Jurídico v1.0.0", body_style))
story.append(Paragraph("Plataformas soportadas: Windows 10+ (64 bits), macOS 10.15+", body_style))
story.append(Spacer(1, 0.3*inch))

story.append(Paragraph("<b>Gracias por usar Sistema Jurídico!</b>", heading2_style))

# Construir el PDF
doc.build(story)
print(f"PDF creado exitosamente: {pdf_path}")
